prov/gni/src/gnix_rma.c

```diff
@@ -1352,8 +1352,10 @@ ssize_t _gnix_rma(struct gnix_fid_ep *ep, enum gnix_fab_req_type fr_type,
 	}
 
 	/* Adding FI_FENCE to an FI_MORE list will break the FI_MORE Chain.
-	 * Remove FI_MORE flag when FI_FENCE flag is present. */
-	if (flags & FI_FENCE) {
+	 * Current FI_MORE implementation does not create remote CQ events.
+	 * Remove FI_MORE flag when FI_FENCE or FI_REMOTE_CQ_DATA flags are
+	 * present. */
+	if ((flags & FI_FENCE) || (flags & FI_REMOTE_CQ_DATA)) {
```

Comment:
Just rebased and i'm not seeing any issues. 

However I tried the following change and some corresponding code to remove this case from being queued on the FI_MORE list
```
- if ((flags & FI_FENCE) || (flags & FI_REMOTE_CQ_DATA)) {
+       if ((flags & FI_FENCE) || (flags & FI_REMOTE_CQ_DATA) ||
+           (ep->vc->peer_caps & FI_RMA_EVENT)) {
                flags &= ~FI_MORE;
        }
```
But I'm observing failures on a lot of gnitests afterwards. Trying to figure out why that might be happening...